### PR TITLE
SCHEMA: Uniformize rules

### PR DESCRIPTION
**Pull request by effigies**
This PR aims to achieve consistency throughout the schema.

* [x] Use consistent syntax for issues (`{issue: {code: ..., message: ..., level: ...}}`)
* [x] Use requirement level as a bare string on fields sidecar rules unless there are additional subfields
* [x] Eliminate redundant checks when encountered
* [x] Conform modality strings (currently have `"mri"` and `"MRI"`
* [x] Conform all selector and check expressions according to the current state of #1080

### CODE REVIEW

File: src/schema/meta/context.yaml [hunk 1]
```diff
@@ -36,6 +39,9 @@ context:
         ignored:
           description: 'Set of ignored files'
           type: array
+        datatypes:
+          description: 'Data types present in the dataset'
+          type: array
```
Comment (effigies) [inline src/schema/meta/context.yaml:44]: Two changes to the context were required. Is this version-bump justifying?

**Comment by effigies** [comment]
I believe this is good to go. Nothing currently backwards incompatible, but some additions motivated by a couple of checks.

**Comment by effigies** [comment]
Bump @rwblair.

**Comment by rwblair** [comment]
Haven't merged any of the function injections into the context eval space so most of these will fail in schema prototype validator. No need to put the cart in front of the horse if our expression language is done changing, I'd be fine with merging these right now.

Should there be a version bump?

**Comment by effigies** [comment]
> Haven't merged any of the function injections into the context eval space so most of these will fail in schema prototype validator. No need to put the cart in front of the horse if our expression language is done changing, I'd be fine with merging these right now.

However works best for you. I could factor out the expression changes, if you'd rather those be a separate PR.

> Should there be a version bump?

Probably. I'll do that.

**Review by rwblair — APPROVED** [review]

**Comment by sappelhoff** [comment]
Will the NIRS branch need to be updated with changes introduced here? If yes, would you mind going the extra mile and taking that on @effigies?

**Comment by effigies** [comment]
`master` merged cleanly into `bep030`. Looking at the PR, I don't see any of the violations I was cleaning up in here.

**Pull request merged**